src/main/java/org/broadinstitute/hellbender/engine/GATKTool.java

```diff
@@ -547,10 +584,43 @@ protected void onStartup() {
      * and so are not sensitive to contig ordering issues).
      */
     private void validateSequenceDictionaries() {
+
         final SAMSequenceDictionary refDict = hasReference() ? reference.getSequenceDictionary() : null;
         final SAMSequenceDictionary readDict = hasReads() ? reads.getSequenceDictionary() : null;
         final List<SAMSequenceDictionary> featureDicts = hasFeatures() ? features.getAllSequenceDictionaries() : Collections.emptyList();
 
+        // Check the master dictionary against the reference / reads / features
+        if (masterSequenceDictionary != null) {
+
+            // Check against the reads
+            if ( hasReads() ) {
+                // When overriding the master sequence dictionary and working with CRAM files, we need to make sure that the
+                // reference we're using is completely contained by the sequence dictionary (i.e. the sequence dictionary is
+                // equal to the reference OR the sequence dictionary is a superset of the reference).
+                // This is accomplished by the call to validateCRAMDictionaryAgainstReference.
+
+                final boolean requireMasterDictionaryIsSuperSet = hasCramInput();
+
+                SequenceDictionaryUtils.validateDictionaries("master sequence dictionary", masterSequenceDictionary,
+                        "reads", readDict, requireMasterDictionaryIsSuperSet, false);
+
+                if ( hasReference() ) {
+                    SequenceDictionaryUtils.validateDictionaries("master sequence dictionary", masterSequenceDictionary,
+                            "reference", refDict, requireMasterDictionaryIsSuperSet, false);
+                }
```

Comment:
This `if ( hasReference() )` block should be outside the `if ( hasReads() )` block, and should replace the other `if (hasReference()) {` block below, which is now redundant. You'll also need to pull the `final boolean requireMasterDictionaryIsSuperSet = hasCramInput();` outside of the `if ( hasReads() ) {` block as well.